Detail page

raymondlf opened the pull request:
@runongirlrunon 
route file setup for detail page -> url: /media/<category>/<MediaID>.
Controller function getDetail() setup for framework MVC testing.
Model MediaTable.php function findByID setup for framework MVC testing.
View get_detail.ctp setup for framework MVC testing.

Rd-Feng committed:
add homeSearch function that search by media title and media category (optional)

Rd-Feng committed:
routes setup, bunch of debug and test code in MediaController, Media table model, and get_detail view.

runongirlrunon commented on src/Template/Media/get_detail.ctp:
```diff
@@ -0,0 +1,141 @@
+<?php
```
Rather than creating a new file, I would repurpose the `detail.ctp` file by moving it into the `Media` folder and adding the raw content (minus any CSS) from this file into that one. You'll see that one already has the default layout attached, which will get the menu bar, so all you need are lines 95-end put into the `<body>` section.

runongirlrunon commented on src/Model/Table/MediaTable.php:
```diff
@@ -81,4 +81,48 @@ public function validationDefault(Validator $validator)
+     * if category is empty/default, search by title only.
+     * Return all media found.
+     */
+    public function homeSearch(Query $query, array $options){
```
I would move these custom searches into the `MediaController`, rather than the Model.

raymondlf replied:
I would suggest that we let the models handle all the functions because cakePHP controller automatically calls functions in models. It seems that cakePHP want controllers to only focus on controlling behaviors (i.e., controller tell models "what" to do and models handle "how" to do).

Rd-Feng committed:
tempCommit

Rd-Feng committed:
Clean up comment.
Re configure url to http://<base url>/media/detail/<MediaID>
All navigation buttons in get_detail.ctp view now point to home.ctp.

Rd-Feng committed:
working detail page

Rd-Feng committed:
abort changes in master
Merge branch 'master' of https://github.com/summer17-csc648-team1/warehausmedia

Rd-Feng committed:
done migrations.

Rd-Feng committed:
Merge branch 'detailPage'

Rd-Feng committed:
user default layout

Commit:
Merge branch 'master' into detailPage

Commit:
minor typo fixes

Commit:
clean up some styling on the detail page

runongirlrunon commented:
@raymondlf I made a few changes to clean things up, which you can view on the last commit I made. LGTM!

runongirlrunon commented:
This closes https://github.com/summer17-csc648-team1/warehausmedia/issues/11.
